modules/platforms/cpp/ignite/client/sql/sql.h

```diff
@@ -44,49 +45,57 @@ class sql {
     sql() = delete;
 
     /**
-     * Executes single SQL statement asynchronously and returns rows.
+     * Executes a single SQL statement asynchronously and returns rows.
      *
      * @param tx Optional transaction. If nullptr implicit transaction for this single operation is used.
+     * @param token Cancellation token. Can be @c nullptr.
      * @param statement Statement to execute.
      * @param args Arguments for the statement (can be empty).
      * @param callback A callback called on operation completion with SQL result set.
      */
-    IGNITE_API void execute_async(transaction *tx, const sql_statement &statement, std::vector<primitive> args,
-        ignite_callback<result_set> callback);
+    IGNITE_API void execute_async(transaction *tx, cancellation_token *token, const sql_statement &statement,
+        std::vector<primitive> args, ignite_callback<result_set> callback);
```

Comment:
Should we for convenience also provide overloads without the `cancellation_token`? In the tests we usually just pass `nullptr`, but that isn't really descriptive. (Same of course also goes for the other functions.)

Reply:
This issue should probably be addressed in another way, because right now we have like 5 arguments, 4 of which can be optional independently of each other. That's a hell of overloads. I prefer to live it like this until we come up with a common solution for cases like this - we can always add overloads in future.